core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/actors/mobs/Necromancer.java

```diff
@@ -225,9 +227,35 @@ public void onZapComplete(){
 			}
 			
 			Buff.affect(mySkeleton, Adrenaline.class, 3f);
+		}*/
+
+
+		if (sprite.visible || mySkeleton.sprite.visible) {
+			sprite.parent.add(new Beam.HealthRay(sprite.center(), mySkeleton.sprite.center()));
+		}
+
+		mySkeleton.HP = Math.min(mySkeleton.HP + mySkeleton.HT/5, mySkeleton.HT);
+		if (mySkeleton.sprite.visible) {
+			mySkeleton.sprite.showStatusWithIcon( CharSprite.POSITIVE, Integer.toString( mySkeleton.HT/5 ), FloatingText.HEALING );
+		}
+
+		if (mySkeleton.buff(Adrenaline.class) != null) {
+			Buff.prolong(mySkeleton, Adrenaline.class, 5f);
 		}
+		else if (mySkeleton.buff(Haste.class) != null) {
+			Buff.detach(mySkeleton, Haste.class);
+			Buff.affect(mySkeleton, Adrenaline.class, 3f);
+		}
+		else if (mySkeleton.buff(Stamina.class) != null) {
+			Buff.detach(mySkeleton, Stamina.class);
+			Buff.affect(mySkeleton, Haste.class, 3f);
+		}
+		else {
+			Buff.affect(mySkeleton, Stamina.class, 3f);
+		}
+
 
-		polished.zapCooldown = Random.NormalIntRange(2, 2);
+		polished.zapCooldown = Random.NormalIntRange(1, 1);
```

Comment:
```suggestion
		polished.zapCooldown = 1
```
Any reason you don't just make it a constant number? There's no reason to pass it through a function